workers/unity/Assets/Gdk/TestUtils/Tests/Editmode/HybridGdkSystemTests.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using JetBrains.Annotations;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils.EditmodeTests
+{
+    public class HybridSystemTests
+    {
+        private struct TestPreparation : IComponentData
+        {
+            public int Value;
+        }
+
+        [UsedImplicitly]
+        [DisableAutoCreation]
+        private class ExampleHybridSystem : ComponentSystem
+        {
+#pragma warning disable 649
+            private struct PreparationData
+            {
+                public int Length;
+
+                // The ComponentArray<> here makes this a hybrid system.
+                public ComponentArray<Rigidbody> GameObjectRigidBody;
+
+                public ComponentDataArray<TestPreparation> PreparationStruct;
+            }
+
+            [Inject] private PreparationData testDataToPrepare;
+#pragma warning restore 649
+
+            public void TestInjection(World world)
+            {
+                OnBeforeCreateManagerInternal(world, 10);
+            }
+
+            public void ManualDispose()
+            {
+                OnBeforeDestroyManagerInternal();
+                OnDestroyManager();
+                OnAfterDestroyManagerInternal();
+            }
+
+            protected override void OnUpdate()
+            {
+                for (var i = 0; i < testDataToPrepare.Length; ++i)
+                {
+                    var preparation = testDataToPrepare.PreparationStruct[i];
+
+                    preparation.Value = 1;
+
+                    testDataToPrepare.PreparationStruct[i] = preparation;
+                }
+            }
+        }
+
+        [TestFixture]
+        private class FixtureNotImplementingHybridTestBase
+        {
+            // If this test fails, then we can get rid of HybridGdkSystemTestBase!
+            [Test]
+            public static void
+                HybridSystems_will_result_in_errors_when_created_if_fixture_does_not_extend_HybridGdkSystemTestBase()
+            {
+                using (var world = new World("test-world"))
+                {
+                    var system = new ExampleHybridSystem();
+
+                    var argumentException = Assert.Throws<ArgumentException>(() =>
+                        {
+                            // This is basically the same action as world.GetOrCreateManager<ExampleHybridSystem>();
+                            // However, if we actually called the above it would cause memory leaks and make other tests fail.
```

Comment:
Out of curiosity, why would it cause mem leaks?

Reply:
Hehe...

In `Unity.Entities.ScriptBehaviourManager`:

```cs
    internal void CreateInstance(World world, int capacity)
    {
      this.OnBeforeCreateManagerInternal(world, capacity);
      try
      {
        this.OnCreateManager(capacity);
        Type type = this.GetType();
        this.sampler = CustomSampler.Create(string.Format("{0} {1}", (object) world.Name, (object) type.FullName));
      }
      catch
      {
        this.OnBeforeDestroyManagerInternal();
        this.OnAfterDestroyManagerInternal();
        throw;
      }
    }
```

The Injection error happens within `OnBeforeCreateManagerInternal`, which means `OnBeforeDestroy` et al will not get called 😓